osm2streets/src/lanes/classic.rs

```diff
@@ -447,11 +473,37 @@ fn add_sidewalks_and_shoulders(
     // For now, model that by putting shoulders.
     if cfg.inferred_sidewalks || tags.is(osm::HIGHWAY, "living_street") {
         if need_fwd_shoulder {
-            fwd_side.push(fwd(&tags, LaneType::Shoulder));
+            fwd_side.push(fwd(highway_type, LaneType::Shoulder));
         }
         if need_back_shoulder {
-            back_side.push(back(&tags, LaneType::Shoulder));
+            back_side.push(back(highway_type, LaneType::Shoulder));
+        }
+    }
+}
+
+fn apply_turn_restrictions(list: &mut Vec<LaneSpec>, value: &str) {
```

Comment:
I adapted this from https://github.com/a-b-street/abstreet/blob/ede907e23971427413b25982ef2c22e7c1309c6a/map_model/src/objects/lane.rs#L232, which does parsing last minute and has some special handling for buses.